spec.md

```diff
@@ -758,6 +759,24 @@ The `code` field MUST be one of the following:
 | code-13 | `UNSUPPORTED`           | the operation is unsupported                               |
 | code-14 | `TOOMANYREQUESTS`       | too many requests                                          |
 
+#### Warnings
+
+Registry implementations MAY include informational warnings in `Warning` headers, as described in https://www.rfc-editor.org/rfc/rfc7234#section-5.5.
+
+If included, `Warning` headers MUST specify a `warn-code` of `299` and a `warn-agent` of `-`, and MUST NOT specify a `warn-date` value.
+
+A registry MUST NOT send more than 4096 bytes of warning data.
```

Comment:
Is this limit per header, or the sum of all headers in a response?

Reply:
for kubernetes, it was the sum... we were paranoid about various proxies or clients setting small limits on how much header content they would accept and breaking responses. The most conservative defaults I found ranged from 8kb-32kb, so I halved the most conservative default I found and capped warning headers to that.

Reply:
```suggestion
A registry MUST NOT send more than 4096 bytes of warning data from all headers combine.
```